magicctapipe/io/io.py

```diff
@@ -0,0 +1,941 @@
+    event_data.query(f"event_type == {EventType.SUBARRAY.value}", inplace=True)
+
+    # Exclude poorly reconstructed events
+    event_data.dropna(
+        subset=["intensity", "time_gradient", "alt_tel", "az_tel"], inplace=True
+    )
+
+    # Exclude the events with duplicated event IDs.
+    # ToBeChecked: if it still happens in recently processed data
+    event_ids, counts = np.unique(
+        event_data.index.get_level_values("event_id"), return_counts=True
+    )
+
+    if np.any(counts > 1):
+        event_ids_dup = event_ids[counts > 1].tolist()
+        event_data.query(f"event_id != {event_ids_dup}", inplace=True)
+
+        logger.warning(
+            f"WARNING: Excluded the events with duplicated event IDs: {event_ids_dup}"
+        )
+
+    logger.info(f"LST-1: {len(event_data)} events")
+
+    # Rename the columns
+    event_data.rename(
+        columns={
+            "delta_t": "time_diff",
+            "alt_tel": "pointing_alt",
+            "az_tel": "pointing_az",
+            "leakage_pixels_width_1": "pixels_width_1",
+            "leakage_pixels_width_2": "pixels_width_2",
+            "leakage_intensity_width_1": "intensity_width_1",
+            "leakage_intensity_width_2": "intensity_width_2",
+            "time_gradient": "slope",
+        },
+        inplace=True,
+    )
+
+    # Change the units of parameters
+    optics = pd.read_hdf(input_file, key="configuration/instrument/telescope/optics")
+    focal_length = optics["equivalent_focal_length"][0]
+
+    event_data["length"] = focal_length * np.tan(np.deg2rad(event_data["length"]))
+    event_data["width"] = focal_length * np.tan(np.deg2rad(event_data["width"]))
+
+    event_data["phi"] = np.rad2deg(event_data["phi"])
+    event_data["psi"] = np.rad2deg(event_data["psi"])
+
+    # Read the subarray description
+    subarray = SubarrayDescription.from_hdf(input_file)
+
+    if focal_length == NOMINAL_FOCLEN_LST:
+        # Set the effective focal length to the subarray
+        subarray.tel[1].optics.equivalent_focal_length = EFFECTIVE_FOCLEN_LST
+        subarray.tel[1].camera.geometry.frame = CameraFrame(
+            focal_length=EFFECTIVE_FOCLEN_LST
+        )
+
+    return event_data, subarray
+
+
+def load_magic_dl1_data_files(input_dir):
+    """
+    Loads MAGIC DL1 data files for the event coincidence with LST-1.
+
+    Parameters
+    ----------
+    input_dir: str
+        Path to a directory where input MAGIC DL1 data files are stored
+
+    Returns
+    -------
+    event_data: pandas.core.frame.DataFrame
+        Data frame of MAGIC events
+    subarray: ctapipe.instrument.subarray.SubarrayDescription
+        MAGIC subarray description
+
+    Raises
+    ------
+    FileNotFoundError
+        If any DL1 data files could not be found in the input directory
+    """
+
+    # Find the input files
+    file_mask = f"{input_dir}/dl1_*.h5"
+
+    input_files = glob.glob(file_mask)
+    input_files.sort()
+
+    if len(input_files) == 0:
```

Comment:
similar check can be also introduced for reading LST1 data in the function above